diff -rupN /tmp/initng-0.6.10.2/plugins/ngc4/libngcclient.c ./plugins/ngc4/libngcclient.c
--- /tmp/initng-0.6.10.2/plugins/ngc4/libngcclient.c	2017-01-06 18:13:27.134675063 +0100
+++ ./plugins/ngc4/libngcclient.c	2017-01-06 17:49:43.006707000 +0100
@@ -123,7 +123,7 @@ reply *ngcclient_send_command(const char
 {
 	read_header header;
 	reply *rep;
-	int sock = -1;
+	int sock = -1, trials = 0;
 
 	/* clear structure just in case */
 	memset(&header, 0, sizeof(read_header));
@@ -145,8 +145,19 @@ reply *ngcclient_send_command(const char
 
 	/*print_out("Sending: %c, %s, %s\r\n", c, l ,o); */
 
-	/* open the socket two way to initng */
-	if ((sock = ngcclient_open_socket(path)) < 1)
+	for (;;)
+	{
+		/* open the two way socket to initng */
+		sock = ngcclient_open_socket (path);
+		if (sock || ++trials > 1)
+			break;
+
+		/* Send SIGHUP signal to initng to recreate /dev/initng */
+		kill (1, SIGHUP);
+		usleep (50000);
+	}
+
+	if (sock < 1)
 	{
 		/*
 		 * Set in ngcclient_open_socket()
